h5netcdf/core.py

```diff
@@ -557,11 +558,11 @@ def _create_dim_scales(self):
                 if self._dim_sizes[dim] is None:
                     kwargs["maxshape"] = (None,)
                 self._h5group.create_dataset(
-                    name=dim, shape=(size,), dtype="S1", **kwargs
+                    name=dim, shape=(size,), dtype=">f4", **kwargs
```

Comment:
To clarify: these other dtype changes are unrelated to the vlen string stuff, right? I assume these are for consistency with netCDF?

Reply:
Yes, I tried to be as consistent with netcdf4 created files.